pkg/login/flow.go

```diff
@@ -167,29 +167,39 @@ func (f *Flow) triggerMethod(provider *Provider) (acsToken string, err error) {
 				if provider.ID == DCOSOIDCAuth0 {
 					loginServer, err = loginserver.New(startFlowURL)
 					if err != nil {
-						return "", err
+						f.logger.Error(err)
+					} else {
+						startFlowURL = loginServer.StartFlowURL()
+						go loginServer.Start()
+						defer loginServer.Close()
 					}
-					startFlowURL = loginServer.StartFlowURL()
-					go loginServer.Start()
 				}
 
 				if err := f.openBrowser(startFlowURL); err != nil {
-					return "", err
-				}
+					// Being unable to open the browser is not critical, a message was prompted
+					// to the user with the URL they should go to in order to start the login flow.
+					f.logger.Info(err)
 
-				if loginServer != nil {
+					// However we close and don't use the web server in this scenario, as the CLI might
+					// be running on a non-desktop environment so the we server would be stuck waiting.
+					// Falling back to reading from STDIN is safer here.
+					//
+					// See https://jira.mesosphere.com/browse/DCOS_OSS-5591
+					loginServer.Close()
+				} else if loginServer != nil {
```

Comment:
Do we have a test for this condition?

Reply:
Not right now, the login server is tested individually already, I'll see if I can add coverage here as well.

Reply:
I've added a test